tests/datasource/data_connector/test_configured_asset_azure_data_connector.py

```diff
@@ -0,0 +1,1576 @@
+       datasource_name: test_environment
+       execution_engine:
+           class_name: PandasExecutionEngine
+       container: my_container
+       name_starts_with: ""
+       assets:
+           TestFiles:
+       default_regex:
+           pattern: (.+)_(.+)_(.+)\\.csv
+           group_names:
+               - name
+               - timestamp
+               - price
+       sorters:
+           - orderby: asc
+             class_name: LexicographicSorter
+             name: name
+           - datetime_format: "%Y%m%d"
+             orderby: desc
+             class_name: DateTimeSorter
+             name: timestamp
+           - orderby: desc
+             class_name: NumericSorter
+             name: price
+       azure_options:
+           account_url: my_account_url.blob.core.windows.net
+           credential: my_credential
+     """,
+    )
+
+    mock_list_keys.return_value = [
+        "alex_20200809_1000.csv",
+        "eugene_20200809_1500.csv",
+        "james_20200811_1009.csv",
+        "abe_20200809_1040.csv",
+        "will_20200809_1002.csv",
+        "james_20200713_1567.csv",
+        "eugene_20201129_1900.csv",
+        "will_20200810_1001.csv",
+        "james_20200810_1003.csv",
+        "alex_20200819_1300.csv",
+    ]
+
+    my_data_connector: ConfiguredAssetAzureDataConnector = (
+        instantiate_class_from_config(
+            config=my_data_connector_yaml,
+            runtime_environment={
+                "name": "general_azure_data_connector",
+                "datasource_name": "test_environment",
+            },
+            config_defaults={
+                "module_name": "great_expectations.datasource.data_connector"
+            },
+        )
+    )
+
+    self_check_report = my_data_connector.self_check()
+
+    assert self_check_report["class_name"] == "ConfiguredAssetAzureDataConnector"
+    assert self_check_report["data_asset_count"] == 1
+    assert self_check_report["data_assets"]["TestFiles"]["batch_definition_count"] == 10
+    assert self_check_report["unmatched_data_reference_count"] == 0
+
+    sorted_batch_definition_list = (
+        my_data_connector.get_batch_definition_list_from_batch_request(
+            BatchRequest(
+                datasource_name="test_environment",
+                data_connector_name="general_azure_data_connector",
+                data_asset_name="TestFiles",
+            )
+        )
+    )
+    assert sorted_batch_definition_list == expected_batch_definitions_sorted
+
+
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.BlobServiceClient"
+)
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.list_azure_keys"
+)
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+def test_return_all_batch_definitions_sorted_sorter_named_that_does_not_match_group(
```

Comment:
Does not match group -- or does match group?  Also, it would be nice to know which error message we are expecting when using `pytest.raises()`.

Reply:
I've updated both! Let me know your thoughts on the updates.